pkg/component/monitoring/prometheus/vpa.go

```diff
@@ -43,11 +44,15 @@ func (p *prometheus) vpa() *vpaautoscalingv1.VerticalPodAutoscaler {
 			},
 			ResourcePolicy: &vpaautoscalingv1.PodResourcePolicy{
 				ContainerPolicies: []vpaautoscalingv1.ContainerResourcePolicy{
+					{
+						ContainerName:    "*",
```

Comment:
Does VPA merge `ContainerPolicies`? Otherwise, there won't be a match. Afaik prometheus runs 2 container only (prometheus, config-reloader) which have already dedicated policies.
I suggest to include this setting into `prometheus` container, `config-reloader` is not scaled anyway.

Reply:
Interesting, we just talked about this a few hours ago with @voelzmo 😄 Actually, we don't know for sure. I was just reflecting the change from https://github.com/gardener/gardener/pull/9175 so that it also gets applied when Prometheus is managed via `prometheus-operator`. In the scope of this PR, this is correct/what we should do. If we think this VPA definition is wrong, then let's fix it separately (my goal is just to translate/refactor the Helm charts to "mgmt via prom-operator", not to change/add/remove/reinvent things). :) OK?

Reply:
I'm pretty sure now, that VPA does not merge ContainerPolicy ([ref](https://github.com/kubernetes/autoscaler/blob/a3c8978e119f56a90a6f8406e76685c21cfecd51/vertical-pod-autoscaler/pkg/utils/vpa/api.go#L186-L201)).
Anyway, we can fix it in an own PR too.

Reply:
Thanks for checking @oliver-goetz!
@voelzmo @nickytd Would you like to open a PR that addresses these concerns?

Reply:
done: https://github.com/gardener/gardener/pull/9192